### ISSUE TITLE
loop optimization

### ISSUE THREAD

**Issue by realmahd1** [mohammad-hassani/iran-city#1]
Hi, Hope you have good times.
When we have a long list of items that we need to loop throw them it's good to optimize our loop.
one important thing you can do to speed up your loop is instead of putting your array length in your for loop like this:

```javascript
for(let i = 0;  i < arr.length ; i++){ 

}
```

change it to this: 

```javascript
const len = arr.length;  

for(let i = 0 ; i < len; i++){

}
```
When you are putting `arr.length`  in for loop its counts array length every time we start looping and it takes time.
instead, we can define our array length as constant and give that constant to for.

You can see the difference between loops in js:

![test_performance](https://user-images.githubusercontent.com/76115755/178098190-fde553f7-0056-4ac4-b548-ab47eeeb1ac8.png)

![jsbench](https://user-images.githubusercontent.com/76115755/178098196-94e72bc9-c8fc-4133-a187-4b2bcd2af361.png)

**Comment by mohammad-hassani** [comment]
@realmahd1 
Hi, thank you for your complete explanation. I will definitely correct it in the next version :ok_hand:

**Comment by mohammad-hassani** [comment]
The Issue has been fixed in #3
